text/0800-ts-adoption-plan.md

```diff
@@ -0,0 +1,833 @@
+
+    - there *are* no new errors related to their use of Ember code
+
+    **OR**
+
+    - any errors which *do* appear represent real possible bugs in your *use* of Ember code
+
+This maximizes the value offered by TypeScript for consumers of Ember’s types (as well as maximizing the benefit to Ember itself internally). Developers who want to  take advantage of the strictest type checking options available to prevent runtime errors will always be able to do so. Accordingly, we do *not* guarantee that code will type check in looser modes, especially when `strictNullChecks` is disabled. *Most* code will still work correctly, but advanced types features used in Ember sometimes rely on the ability to distinguish between `null` or `undefined` and other types, and we do not commit to avoiding those—not least because there is otherwise an exponential explosion of possible strictness combinations we would have to test.
+
+<details><summary>On including <code>noUncheckedIndexedAccess</code></summary>
+
+We include the `noUncheckedIndexedAccess` setting in our recommended settings because it catches one of the most significant holes in the TypeScript type system prior to its introduction: not accounting for the fact that "index"-style access to objects and arrays *always* allows arbitrary access to any index (numeric or string), whether or not it is defined. For example, consider this type (not recommended for *other* reasons, but legal):
+
+```ts
+type Anything {
+  [key: string]: string;
+}
+
+function describe(anything: Anything) {
+  let email = person['email'];
+}
+```
+
+Here, `email` would be typed as `string`, and so TypeScript would not complain if you did something like `email.length`. But you can *call* this with an object which does *not* have an `email` field on it:
+
+```ts
+describe({ noEmailsHere: 'bwahahaha' });
+```
+
+This will cause a runtime error instead of a type error. However, TypeScript is (as of recent releases) smart enough to understand that if you *check* for `"email"` it can be *known* to be present:
+
+```ts
+function describe(anything: Anything) {
+  if (anything['email']) {
+    let email = anything['email']
+    email.length
+  }
+}
+```
+
+Accordingly, we include this as part of Ember’s own strictness settings and defaults generated for users.[^should-be-strict]
+
+</details>
+
+[^should-be-strict]: The Typed Ember team thinks TypeScript *should* have included this under `strict: true`!
+
+
+#### Primary packages
+
+The primary packages in the ecosystem, `ember-source`, `ember-data`, and `ember-cli`, will adopt the [rolling window](https://github.com/chriskrycho/ember-rfcs/blob/semver-for-ts/text/0730-semver-for-ts.md#rolling-support-windows) policy:
```

Comment:
Will update to that effect. The one thing I would say (that we should decide on together) is: some of those actually currently support *more* than Ember itself does, and for that reason I think it would be fine for them to adopt a “simple majors” policy with a caveat that they will never support *less* than Ember itself does.